libs/features/personalization/personalization.js

```diff
@@ -329,70 +332,84 @@ function registerInBlockActions(cmd, manifestId, targetManifestId) {
+            // eslint-disable-next-line no-param-reassign
+            selector = selector.replace(simplifiedSelector, 'p strong a');
+            break;
+          case simplifiedSelector.includes('secondary-cta'):
+            // eslint-disable-next-line no-param-reassign
+            selector = selector.replace(simplifiedSelector, 'p em a');
+            break;
+          case simplifiedSelector.includes('action-area'):
+            // eslint-disable-next-line no-param-reassign
+            selector = selector.replace(simplifiedSelector, 'p:has(em a, strong a)');
+            break;
+          default: break;
+        }
+      });
+    });
+    // transalte custom block selectors to CSS selectors
+    const customBlockSelectors = selector.match(/\.\w+-?\w+\d+/g);
+    customBlockSelectors?.forEach((customBlockSelector) => {
+      const n = customBlockSelector.match(/\d+/g);
+      const blockName = customBlockSelector.replace(/(\.|\d+)/g, '');
+      const cssOptimizedSelector = ` .${blockName}:nth-child(${n} of .${blockName})`;
+      // eslint-disable-next-line no-param-reassign
+      selector = selector.replace(customBlockSelector, cssOptimizedSelector);
+    });
+    // eslint-disable-next-line no-param-reassign
+    selector = rootEl === document ? `body > main ${selector}` : `:scope ${selector}`;
+    const element = querySelector(rootEl || document, selector);
 
-  let section = null;
-  if (terms[0]?.startsWith('section')) {
-    const sectionIdx = getTrailingNumber(terms[0]) || 1;
-    section = getSection(rootEl, sectionIdx);
-    terms.shift();
-  }
+    // TODO: for testing purposes only. Remove when done
+    console.log('=====================================');
+    console.log('selector: ', originalSelector, ' ==> ', selector);
+    console.log('element: ', element ? 'found' : '!!! NOT FOUND !!!');
 
-  if (terms.length) {
-    const blockStr = terms.shift();
-    if (blockStr.includes('.')) {
-      selectedEl = querySelector(section || rootEl, blockStr);
-    } else {
-      const blockIndex = getTrailingNumber(blockStr) || 1;
-      const blockName = blockStr.replace(`${blockIndex}`, '');
-      if (blockName === 'block') {
-        if (!section) section = getSection(rootEl, 1);
-        selectedEl = querySelector(section, `:scope > div:nth-of-type(${blockIndex})`);
-      } else {
-        selectedEl = querySelector(section || rootEl, `.${blockName}`, true)?.[blockIndex - 1];
-      }
-    }
-  } else if (section) {
-    return section;
-  }
-
-  if (terms.length) {
-    // find targeted table cell in rowX colY format
-    const rowColMatch = /row(?<row>\d+)\s+col(?<col>\d+)/gm.exec(terms.join(' '));
-    if (rowColMatch) {
-      const { row, col } = rowColMatch.groups;
-      const tableCell = querySelector(selectedEl, `:nth-child(${row}) > :nth-child(${col})`);
-      if (!tableCell) return null;
-      return getDivInTargetedCell(tableCell);
-    }
+    return element;
```

Comment:
This assumes you send action into the function again.
```suggestion
    if (action.includes('pendtosection') && element?.parentNode?.nodeName) return;
    return element;
```

Reply:
not sure what this one means . If I add this code the tests for append/prepend section breaks, because `getSelectedElement` will just return nothing